### ISSUE TITLE
How to generate correct RSA key which can be used in Linux kernel space?

### ISSUE THREAD

**Issue by XinShuoWang** [alekseymmm/lk_crypto_test#1]
Hi, I saw your code for this project, and I think it's great. I have a question in the process of using it, I wonder if you can answer it: I want to replace the key pair in `lk_crypto_test/kern/rdx_crypto.c` with the **DER** format (a subset of the BER format) generated by OpenSSL, but it is not successful, it will make an error at the step `crypto_akcipher_set_pub_key`. I want to know how you generated the key pair in the code, thanks!

**Comment by alekseymmm** [comment]
Hi @XinShuoWang .
I am really surprised that someone found this project. I hope I can help even though it has been a while since I wrote this code.
The way I generated key pair is described in 
`lk_crypto_test/usr/gen_keys.sh`
the script is very simple and gets one argument - the key size
It generates priv and public keys.

And I have written some useful utility `convert` that convert them to C string that can be copy-pasted into kernel module.

P.S. pay attention to correct string length in kernel module

**Comment by XinShuoWang** [comment]
Thank you very much for answering my questions, I have tried it and it works very well!

**Comment by XinShuoWang** [comment]
Thank you very much for answering my questions, I have tried it and it works very well!


On 3/29/2020 00:53，Aleksei<notifications@github.com> wrote：

Hi @XinShuoWang .
I am really surprised that someone found this project. I hope I can help even though it has been a while since I wrote this code.
The way I generated key pair is described in
lk_crypto_test/usr/gen_keys.sh
the script is very simple and gets one argument - the key size
It generates priv and public keys.

And I have written some useful utility convert that convert them to C string that can be copy-pasted into kernel module.
